### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by atrakh**
Syncs READMEs and CLI documentation with current state of the site docs. Also includes some general improvements to documentation readability.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
-| `commitUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per commit. Example: `https://github.com/launchdarkly/git-flag-parser/commit/${sha}`. Allowed template variables: `branchName`, `sha`. If `commitUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" | no |
-| `hunkUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per code reference. Example: `https://github.com/launchdarkly/git-flag-parser/blob/${sha}/${filePath}#L${lineNumber}`. Allowed template variables: `sha`, `filePath`, `lineNumber`. If `hunkUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" | no |
+macOS users should download the darwin release for their respective system architecture.
+
+The `git-flag-parser` requires [The Silver Searcher](https://github.com/ggreer/the_silver_searcher#installing) to be installed as a dependency, so make sure it has been installed and added to your system PATH before running the git-flag-parser.
```
Comment (vlymar) [inline README.md]: super small nit, I'd prefer either "system's `$PATH`" or "system path"

File: README.md [hunk 2]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+
+A number of command-line arguments are available to the parser, some optional, and some required. Command line arguments may be passed to the program in any order.
+
+### Required arguments
+
+| Option | Description |
+|-|-|
+| `accessToken` | LaunchDarkly [personal access token](https://docs.launchdarkly.com/docs/api-access-tokens) with writer-level access, or access to the `code-references` [custom role](https://docs.launchdarkly.com/v2.0/docs/custom-roles) resource |
```
Comment (vlymar) [inline README.md]: we should have a ticket to update https://docs.launchdarkly.com/v2.0/docs/custom-roles with the code refs resource identifier.

File: README.md [hunk 3]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `dir` | Path to existing checkout of the git repo. If a cloneEndpoint is provided, this option is not required. |
```
Comment (vlymar) [inline README.md]:
```suggestion
| `dir` | **Absolute** path to existing checkout of the git repo. Relative paths are not currently supported. |
```

File: README.md [hunk 4]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `projKey` | A LaunchDarkly project key. |
+| `repoName` | Git repo name. Will be displayed in LaunchDarkly |
 
-## Testing
+Here's an example shell invocation of the git-flag-parser for one of LaunchDarkly's demo repositories:, with the binary located in the current directory, and a minimal configuration:
+
+```shell
+./git-flag-parser \
+  -accessToken="$YOUR_LAUNCHDARKLY_ACCESS_TOKEN" \
+  -dir="path/to/git/repo" \
```
Comment (vlymar) [inline README.md]:
```suggestion
  -dir="/path/to/git/repo" \
```

File: README.md [hunk 5]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+  -repoName="my-repository" \
+  -projKey="default"
+```
+
+### Optional arguments
+
+| Option | Description | Default |
+|-|-|-|
+| `baseUri` | Set the base URL of the LaunchDarkly server for this configuration. Only necessary if using a private instance of LaunchDarkly. | `https://app.launchdarkly.com` |
+| `cloneEndpoint` | If provided, will clone the repo from this endpoint to the provided `dir`. If authentication is required, this endpoint should be authenticated. Supports the https protocol for git cloning. Example: `https://username:password@github.com/username/repository.git` | n/a |
```
Comment (vlymar) [inline README.md]: Might as well remove this now right?

File: README.md [hunk 6]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `contextLines` | The number of context lines to send to LaunchDarkly. If < 0, no source code will be sent to LaunchDarkly. If 0, only the line containing flag references will be sent. If > 0, will send that number of context lines above and below the flag reference. A maximum of 5 context lines may be provided. | -1 |
+| `defaultBranch` | The git default branch. The LaunchDarkly UI will default to display code references for this branch. | "master" |
+| `exclude` | A regular expression defining the files and directories which the flag parser should exclude. | "" |
```
Comment (vlymar) [inline README.md]: I think we should offer some examples and a regex specification. I don't think I even know how to use this correctly, e.g. which if the following is correct: `vendor/`, `vendor/*`, `vendor/.*`?

File: README.md [hunk 7]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `updateSequenceId` | An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing `updateSequenceId` is less than the new `updateSequenceId`. Examples: the time a `git push` was initiated, CI build number, the current unix timestamp. | n/a |
+| `repoHead` | The HEAD or ref to retrieve code references from. Should be provided if the `git push` was initiated on a non-master branch. | "master" | no |
```
Comment (vlymar) [inline README.md]: the wording is a bit confusing here. also, can you remind me why we have this option?

File: README.md [hunk 8]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `repoType` | The repo service provider. Used to correctly categorize repositories in the LaunchDarkly UI. Acceptable values: github\|bitbucket\|custom | "custom" |
```
Comment (vlymar) [inline README.md]: i wonder if we should rename "custom" to "other." I remember someone during secret shopper not understanding the difference, and wondering if they should just use "custom." 

"correctly categorize repositories in the LaunchDarkly UI" is hard to understand. an alternative would be "Used to generate repository links in the LaunchDarkly UI.

File: README.md [hunk 9]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `repoUrl` | The display url for the repository. If provided for a github or bitbucket repository, LaunchDarkly will attempt to automatically generate source code links. Example: `https://github.com/launchdarkly/git-flag-parser` | "" |
+| `commitUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per commit. Example: `https://github.com/launchdarkly/git-flag-parser/commit/${sha}`. Allowed template variables: `branchName`, `sha`. If `commitUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" |
```
Comment (vlymar) [inline README.md]: > If `commitUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types.

Do we actually do this? I would expect that we'd only do it if we get `repoUrl` _and_ `repoType`

File: README.md [hunk 10]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
+| `hunkUrlTemplate` | If provided, LaunchDarkly will attempt to generate links to your Git service provider per code reference. Example: `https://github.com/launchdarkly/git-flag-parser/blob/${sha}/${filePath}#L${lineNumber}`. Allowed template variables: `sha`, `filePath`, `lineNumber`. If `hunkUrlTemplate` is not provided, but `repoUrl` is provided, LaunchDarkly will automatically generate links for github or bitbucket repo types. | "" |
```
Comment (vlymar) [inline README.md]: same question as above

File: README.md [hunk 11]
```diff
@@ -21,26 +21,61 @@ This repository provides solutions for configuring [LaunchDarkly code references
...
 
-Set up your development environment by installing Go and running `make init` to install the linter. To lint and run tests, run `make test`.
+```shell
+./git-flag-parser \
+  -accessToken="$YOUR_LAUNCHDARKLY_ACCESS_TOKEN" \
+  -dir="path/to/git/repo" \
```
Comment (vlymar) [inline README.md]:
```suggestion
  -dir="/path/to/git/repo" \
```

File: TESTING.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+## Testing
+
+**Note: this file pertains to the testing of `git-flag-parser` for those actively developing it. It is not intended for testing the execution fo the parser against a real repository.**
```
Comment (vlymar) [inline TESTING.md]: 👍

File: parse/circleci/orb.yml [hunk 13]
```diff
@@ -73,7 +73,7 @@ jobs:
         type: string
         default: "-1"
       exclude:
-        description:  A regular expression defining the files and directories which the flag parser should exclude.
+        description:  "A regular expression (PCRE) defining the files and directories which the flag parser should exclude. Example: `vendor/`"
```
Comment (vlymar) [inline parse/circleci/orb.yml]: ok, I see that you added the regex standard here. Based on your example, it looks like partial matches are filtered. just to be very explicit, we should say that (unless it actually does a prefix match, in which case we should say that

File: parse/internal/options/options.go [hunk 14]
```diff
@@ -95,12 +95,12 @@ var options = optionMap{
...
+	Exclude:           option{"", `A regular expression (PCRE) defining the files and directories which the flag parser should exclude. Example: "vendor/"`, false},
 	ProjKey:           option{"", "LaunchDarkly project key.", true},
-	UpdateSequenceId:  option{noUpdateSequenceId, "An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing `updateSequenceId` is less than the new `updateSequenceId`. Examples: the time a `git push` was initiated, CI build number, the current unix timestamp.", false},
+	UpdateSequenceId:  option{noUpdateSequenceId, `An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing "updateSequenceId" is less than the new "updateSequenceId". Examples: the time a "git push" was initiated, CI build number, the current unix timestamp.`, false},
```
Comment (vlymar) [inline parse/internal/options/options.go]: i saw this spit out a really weird result for Alexis. it just said `git push`. just want to confirm thats fixed.

File: README.md (hunk 5 above)
Comment (atrakh) [reply README.md]: Remove with https://github.com/launchdarkly/git-flag-parser/pull/33

File: README.md (hunk 9 above)
Comment (atrakh) [reply README.md]: Yes, that's right. The end of that quote says "links for github or bitbucket repo types". I'll update it to be clearer.

File: README.md (hunk 2 above)
Comment (atrakh) [reply README.md]: https://app.clubhouse.io/launchdarkly/story/29783/update-custom-role-docs

File: parse/internal/options/options.go (hunk 14 above)
Comment (atrakh) [reply parse/internal/options/options.go]: I wasn't able to reproduce. Is it possible his terminal wasn't wrapping lines?

File: README.md (hunk 7 above)
Comment (atrakh) [reply README.md]: We have this option so we can decide which branch to checkout to before scanning for code references. I have a separate ticket filed for deciding if we should remove this or not.

File: parse/internal/options/options.go (hunk 14 above)
Comment (vlymar) [reply parse/internal/options/options.go]: that is possible. we should check again next secret shopper

File: parse/internal/options/options.go [hunk 15]
```diff
@@ -95,15 +95,15 @@ var options = optionMap{
...
-	Exclude:           option{"", `A regular expression (PCRE) defining the files and directories which the flag parser should exclude. Example: "vendor/"`, false},
+	Exclude:           option{"", `A regular expression (PCRE) defining the files and directories which the flag parser should exclude. Partial matches are allowed. Examples: "vendor/", "vendor/*`, false},
 	ProjKey:           option{"", "LaunchDarkly project key.", true},
 	UpdateSequenceId:  option{noUpdateSequenceId, `An integer representing the order number of code reference updates. Used to version updates across concurrent executions of the flag parser. If not provided, data will always be updated. If provided, data will only be updated if the existing "updateSequenceId" is less than the new "updateSequenceId". Examples: the time a "git push" was initiated, CI build number, the current unix timestamp.`, false},
-	RepoHead:          option{"master", "The HEAD or ref to retrieve code references from.", false},
+	RepoHead:          option{"master", "he branch to scan for code references. Should be provided if the `git push` was initiated on a non-master branch.", false},
```
Comment (vlymar) [inline parse/internal/options/options.go]:
```suggestion
	RepoHead:          option{"master", "The branch to scan for code references. Should be provided if the `git push` was initiated on a non-master branch.", false},
```

**Pull request merged**